From 1f2258f3645c6ed5eebda50b4b575b67330ec04e Mon Sep 17 00:00:00 2001
From: Calvin Johnson <calvin.johnson@nxp.com>
Date: Wed, 20 Jun 2018 10:22:50 +0530
Subject: [PATCH 18/22] staging: fsl_ppfe/eth: use mask for rx max frame len

Define and use PFE_RCR_MAX_FL_MASK to properly set Rx max frame
length of MAC Receive Control Register.

Signed-off-by: Calvin Johnson <calvin.johnson@nxp.com>

[ Srinidhi Rao : Ported this patch to photon linux from
  qoriq-components linux 'linux-4.14-nxp'
  (https://source.codeaurora.org/external/qoriq/qoriq-components/linux)
]

Signed-off-by: srinidhira0 <srinidhir@vmware.com>
---
 drivers/staging/fsl_ppfe/pfe_hal.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/staging/fsl_ppfe/pfe_hal.c b/drivers/staging/fsl_ppfe/pfe_hal.c
index 0915034b281b..7e89606bb40c 100644
--- a/drivers/staging/fsl_ppfe/pfe_hal.c
+++ b/drivers/staging/fsl_ppfe/pfe_hal.c
@@ -19,6 +19,8 @@
 #include "pfe_mod.h"
 #include "pfe/pfe.h"
 
+#define PFE_RCR_MAX_FL_MASK	0xC000FFFF
+
 void *cbus_base_addr;
 void *ddr_base_addr;
 unsigned long ddr_phys_base_addr;
@@ -1011,8 +1013,8 @@ void gemac_no_broadcast(void *base)
 void gemac_enable_1536_rx(void *base)
 {
 	/* Set 1536 as Maximum frame length */
-	writel(readl(base + EMAC_RCNTRL_REG) | (1536 << 16), base +
-		EMAC_RCNTRL_REG);
+	writel((readl(base + EMAC_RCNTRL_REG) & PFE_RCR_MAX_FL_MASK)
+		| (1536 << 16), base +	EMAC_RCNTRL_REG);
 }
 
 /* GEMAC enable jumbo function.
@@ -1020,8 +1022,8 @@ void gemac_enable_1536_rx(void *base)
  */
 void gemac_enable_rx_jmb(void *base)
 {
-	writel(readl(base + EMAC_RCNTRL_REG) | (JUMBO_FRAME_SIZE << 16), base
-		+ EMAC_RCNTRL_REG);
+	writel((readl(base + EMAC_RCNTRL_REG) & PFE_RCR_MAX_FL_MASK)
+		| (JUMBO_FRAME_SIZE << 16), base + EMAC_RCNTRL_REG);
 }
 
 /* GEMAC enable stacked vlan function.
-- 
2.14.2

